src/context/UserDataContext/UserDataContext.tsx

```diff
@@ -22,6 +22,58 @@ import UserProgressOnProblemsProperty, {
 } from './properties/userProgressOnProblems';
 import LastVisitProperty, { LastVisitAPI } from './properties/lastVisit';
 
+// Object for counting online users
+var Gathering = (function () {
+  function Gathering(databaseReference) {
+    this.db = databaseReference;
+
+    this.room = this.db.ref('gatherings/globe');
+    this.user = null;
+
+    this.join = function (uid) {
+      if (this.user) {
+        console.error('Already joined.');
+        return false;
+      }
+
+      this.user = uid ? this.room.child(uid) : this.room.push();
+
+      // Add user to presence list when online.
+      var presenceRef = this.db.ref('.info/connected');
+      let self = this;
+      presenceRef.on('value', function (snap) {
+        if (snap.val()) {
+          self.user.onDisconnect().remove();
+          self.user.set(true);
+        }
+      });
+      return true;
+    };
+
+    this.leave = function () {
+      this.user.remove();
+    };
+
+    this.over = function () {
+      this.room.remove();
+    };
```

Comment:
Do you think these two functions are necessary?

Reply:
nah idts